{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This notebook compares the outputs from single and multi node runs (only difference between the two is the scheduler, using TOIL to handle workflows across multiple nodes or cwlref-runner to handle all steps in serial). Does 2 things:\n",
    "- Add up the annotated counts and perform a correlation analysis\n",
    "- Open/compare the STAR log files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "input_dir = '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/'\n",
    "output_dir = '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/outputs/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam',\n",
       " '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam',\n",
       " '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam',\n",
       " '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam',\n",
       " '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam',\n",
       " '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam',\n",
       " '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/se_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam',\n",
       " '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/se_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_bams = sorted(glob.glob(os.path.join(input_dir, '*/*.bam')))\n",
    "all_bams"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v1.5.3\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m8 BAM files  \u001b[0m \u001b[0m                                   ||\n",
      "||                           \u001b[32mP\u001b[36m inputs/pe_multi/ENCFF040CTS.fastqTr.sorted\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mP\u001b[36m inputs/pe_multi/ENCFF201PEI.fastqTr.sorted\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mP\u001b[36m inputs/pe_single/ENCFF040CTS.fastqTr.sorte\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mP\u001b[36m inputs/pe_single/ENCFF201PEI.fastqTr.sorte\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mS\u001b[36m inputs/se_multi/ENCFF040CTS.fastqTr.sorted\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mS\u001b[36m inputs/se_multi/ENCFF201PEI.fastqTr.sorted\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mS\u001b[36m inputs/se_single/ENCFF040CTS.fastqTr.sorte\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mS\u001b[36m inputs/se_single/ENCFF201PEI.fastqTr.sorte\u001b[0m ... \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36m/home/bay001/projects/codebase/bulk_rnaseq_p\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||                 Summary : \u001b[36m/home/bay001/projects/codebase/bulk_rnaseq_p\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||              Annotation : \u001b[36m/projects/ps-yeolab3/bay001/annotations/hg19\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||      Dir for temp files : \u001b[36m/home/bay001/projects/codebase/bulk_rnaseq_p\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||         Strand specific : \u001b[36mreversely stranded\u001b[0m \u001b[0m                              ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\=====================\u001b[36m http://subread.sourceforge.net/ \u001b[0m======================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file /projects/ps-yeolab3/bay001/annotations/hg19/genc\u001b[0m ... \u001b[0m||\n",
      "||    Features : \u001b[36m1196293\u001b[0m \u001b[0m                                                     ||\n",
      "||    Meta-features : \u001b[36m57820\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m25\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapp\u001b[0m ... \u001b[0m||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Assign reads to features... \u001b[0m                                            ||\n",
      "||    Total reads : \u001b[36m83504246\u001b[0m \u001b[0m                                                 ||\n",
      "||    Successfully assigned reads : \u001b[36m27263390 (32.6%)\u001b[0m \u001b[0m                         ||\n",
      "||    Running time : \u001b[36m2.44 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapp\u001b[0m ... \u001b[0m||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Assign reads to features... \u001b[0m                                            ||\n",
      "||    Total reads : \u001b[36m96531504\u001b[0m \u001b[0m                                                 ||\n",
      "||    Successfully assigned reads : \u001b[36m29563998 (30.6%)\u001b[0m \u001b[0m                         ||\n",
      "||    Running time : \u001b[36m2.80 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STARUnmap\u001b[0m ... \u001b[0m||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Assign reads to features... \u001b[0m                                            ||\n",
      "||    Total reads : \u001b[36m83504246\u001b[0m \u001b[0m                                                 ||\n",
      "||    Successfully assigned reads : \u001b[36m27263390 (32.6%)\u001b[0m \u001b[0m                         ||\n",
      "||    Running time : \u001b[36m2.44 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STARUnmap\u001b[0m ... \u001b[0m||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Assign reads to features... \u001b[0m                                            ||\n",
      "||    Total reads : \u001b[36m96531504\u001b[0m \u001b[0m                                                 ||\n",
      "||    Successfully assigned reads : \u001b[36m29563998 (30.6%)\u001b[0m \u001b[0m                         ||\n",
      "||    Running time : \u001b[36m2.86 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapp\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Assign reads to features... \u001b[0m                                            ||\n",
      "||    Total reads : \u001b[36m43184147\u001b[0m \u001b[0m                                                 ||\n",
      "||    Successfully assigned reads : \u001b[36m31381140 (72.7%)\u001b[0m \u001b[0m                         ||\n",
      "||    Running time : \u001b[36m1.26 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapp\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Assign reads to features... \u001b[0m                                            ||\n",
      "||    Total reads : \u001b[36m50152182\u001b[0m \u001b[0m                                                 ||\n",
      "||    Successfully assigned reads : \u001b[36m36135255 (72.1%)\u001b[0m \u001b[0m                         ||\n",
      "||    Running time : \u001b[36m1.44 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file inputs/se_single/ENCFF040CTS.fastqTr.sorted.STARUnmap\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Assign reads to features... \u001b[0m                                            ||\n",
      "||    Total reads : \u001b[36m43184147\u001b[0m \u001b[0m                                                 ||\n",
      "||    Successfully assigned reads : \u001b[36m31381140 (72.7%)\u001b[0m \u001b[0m                         ||\n",
      "||    Running time : \u001b[36m1.26 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file inputs/se_single/ENCFF201PEI.fastqTr.sorted.STARUnmap\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Assign reads to features... \u001b[0m                                            ||\n",
      "||    Total reads : \u001b[36m50152182\u001b[0m \u001b[0m                                                 ||\n",
      "||    Successfully assigned reads : \u001b[36m36135255 (72.1%)\u001b[0m \u001b[0m                         ||\n",
      "||    Running time : \u001b[36m1.47 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                         \u001b[36mRead assignment finished.\u001b[0m \u001b[0m                         ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"/home/bay001/projects/c \u001b[0m ||\n",
      "|| odebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/outp \u001b[0m ||\n",
      "|| uts/counts.txt\" \u001b[0m                                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\=====================\u001b[36m http://subread.sourceforge.net/ \u001b[0m======================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "gencode = '/projects/ps-yeolab3/bay001/annotations/hg19/gencode_v19/gencode.v19.annotation.gtf'\n",
    "counts = os.path.join(output_dir, 'counts.txt')\n",
    "\n",
    "cmd = 'module load subreadfeaturecounts;featureCounts '\n",
    "cmd += '-a {} -s 2 -o {} inputs/*/*.bam'.format(gencode, counts)\n",
    "\n",
    "! $cmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/se_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/se_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Geneid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000223972.4</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000227232.4</th>\n",
       "      <td>52</td>\n",
       "      <td>118</td>\n",
       "      <td>52</td>\n",
       "      <td>118</td>\n",
       "      <td>41</td>\n",
       "      <td>86</td>\n",
       "      <td>41</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000243485.2</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000237613.2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000268020.2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "Geneid                                                                                                             \n",
       "ENSG00000223972.4                                                  1                                               \n",
       "ENSG00000227232.4                                                 52                                               \n",
       "ENSG00000243485.2                                                  1                                               \n",
       "ENSG00000237613.2                                                  0                                               \n",
       "ENSG00000268020.2                                                  0                                               \n",
       "\n",
       "                   inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "Geneid                                                                                                             \n",
       "ENSG00000223972.4                                                  0                                               \n",
       "ENSG00000227232.4                                                118                                               \n",
       "ENSG00000243485.2                                                  0                                               \n",
       "ENSG00000237613.2                                                  0                                               \n",
       "ENSG00000268020.2                                                  0                                               \n",
       "\n",
       "                   inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "Geneid                                                                                                              \n",
       "ENSG00000223972.4                                                  1                                                \n",
       "ENSG00000227232.4                                                 52                                                \n",
       "ENSG00000243485.2                                                  1                                                \n",
       "ENSG00000237613.2                                                  0                                                \n",
       "ENSG00000268020.2                                                  0                                                \n",
       "\n",
       "                   inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "Geneid                                                                                                              \n",
       "ENSG00000223972.4                                                  0                                                \n",
       "ENSG00000227232.4                                                118                                                \n",
       "ENSG00000243485.2                                                  0                                                \n",
       "ENSG00000237613.2                                                  0                                                \n",
       "ENSG00000268020.2                                                  0                                                \n",
       "\n",
       "                   inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "Geneid                                                                                                             \n",
       "ENSG00000223972.4                                                  1                                               \n",
       "ENSG00000227232.4                                                 41                                               \n",
       "ENSG00000243485.2                                                  0                                               \n",
       "ENSG00000237613.2                                                  0                                               \n",
       "ENSG00000268020.2                                                  0                                               \n",
       "\n",
       "                   inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "Geneid                                                                                                             \n",
       "ENSG00000223972.4                                                  0                                               \n",
       "ENSG00000227232.4                                                 86                                               \n",
       "ENSG00000243485.2                                                  0                                               \n",
       "ENSG00000237613.2                                                  0                                               \n",
       "ENSG00000268020.2                                                  0                                               \n",
       "\n",
       "                   inputs/se_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "Geneid                                                                                                              \n",
       "ENSG00000223972.4                                                  1                                                \n",
       "ENSG00000227232.4                                                 41                                                \n",
       "ENSG00000243485.2                                                  0                                                \n",
       "ENSG00000237613.2                                                  0                                                \n",
       "ENSG00000268020.2                                                  0                                                \n",
       "\n",
       "                   inputs/se_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \n",
       "Geneid                                                                                                             \n",
       "ENSG00000223972.4                                                  0                                               \n",
       "ENSG00000227232.4                                                 86                                               \n",
       "ENSG00000243485.2                                                  0                                               \n",
       "ENSG00000237613.2                                                  0                                               \n",
       "ENSG00000268020.2                                                  0                                               "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "featurecounts_output = pd.read_csv(counts, sep='\\t', skiprows=1, index_col=0)\n",
    "featurecounts_output = featurecounts_output.iloc[:,5:]\n",
    "featurecounts_output.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/se_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/se_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.977703</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.977703</td>\n",
       "      <td>0.992351</td>\n",
       "      <td>0.972402</td>\n",
       "      <td>0.992351</td>\n",
       "      <td>0.972402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <td>0.977703</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.977703</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.968103</td>\n",
       "      <td>0.984047</td>\n",
       "      <td>0.968103</td>\n",
       "      <td>0.984047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.977703</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.977703</td>\n",
       "      <td>0.992351</td>\n",
       "      <td>0.972402</td>\n",
       "      <td>0.992351</td>\n",
       "      <td>0.972402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <td>0.977703</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.977703</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.968103</td>\n",
       "      <td>0.984047</td>\n",
       "      <td>0.968103</td>\n",
       "      <td>0.984047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <td>0.992351</td>\n",
       "      <td>0.968103</td>\n",
       "      <td>0.992351</td>\n",
       "      <td>0.968103</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.983265</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.983265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <td>0.972402</td>\n",
       "      <td>0.984047</td>\n",
       "      <td>0.972402</td>\n",
       "      <td>0.984047</td>\n",
       "      <td>0.983265</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.983265</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>inputs/se_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <td>0.992351</td>\n",
       "      <td>0.968103</td>\n",
       "      <td>0.992351</td>\n",
       "      <td>0.968103</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.983265</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.983265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>inputs/se_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <td>0.972402</td>\n",
       "      <td>0.984047</td>\n",
       "      <td>0.972402</td>\n",
       "      <td>0.984047</td>\n",
       "      <td>0.983265</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.983265</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           1.000000                                               \n",
       "inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           0.977703                                               \n",
       "inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STA...                                           1.000000                                               \n",
       "inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STA...                                           0.977703                                               \n",
       "inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           0.992351                                               \n",
       "inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           0.972402                                               \n",
       "inputs/se_single/ENCFF040CTS.fastqTr.sorted.STA...                                           0.992351                                               \n",
       "inputs/se_single/ENCFF201PEI.fastqTr.sorted.STA...                                           0.972402                                               \n",
       "\n",
       "                                                    inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           0.977703                                               \n",
       "inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           1.000000                                               \n",
       "inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STA...                                           0.977703                                               \n",
       "inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STA...                                           1.000000                                               \n",
       "inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           0.968103                                               \n",
       "inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           0.984047                                               \n",
       "inputs/se_single/ENCFF040CTS.fastqTr.sorted.STA...                                           0.968103                                               \n",
       "inputs/se_single/ENCFF201PEI.fastqTr.sorted.STA...                                           0.984047                                               \n",
       "\n",
       "                                                    inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           1.000000                                                \n",
       "inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           0.977703                                                \n",
       "inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STA...                                           1.000000                                                \n",
       "inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STA...                                           0.977703                                                \n",
       "inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           0.992351                                                \n",
       "inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           0.972402                                                \n",
       "inputs/se_single/ENCFF040CTS.fastqTr.sorted.STA...                                           0.992351                                                \n",
       "inputs/se_single/ENCFF201PEI.fastqTr.sorted.STA...                                           0.972402                                                \n",
       "\n",
       "                                                    inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           0.977703                                                \n",
       "inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           1.000000                                                \n",
       "inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STA...                                           0.977703                                                \n",
       "inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STA...                                           1.000000                                                \n",
       "inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           0.968103                                                \n",
       "inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           0.984047                                                \n",
       "inputs/se_single/ENCFF040CTS.fastqTr.sorted.STA...                                           0.968103                                                \n",
       "inputs/se_single/ENCFF201PEI.fastqTr.sorted.STA...                                           0.984047                                                \n",
       "\n",
       "                                                    inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           0.992351                                               \n",
       "inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           0.968103                                               \n",
       "inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STA...                                           0.992351                                               \n",
       "inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STA...                                           0.968103                                               \n",
       "inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           1.000000                                               \n",
       "inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           0.983265                                               \n",
       "inputs/se_single/ENCFF040CTS.fastqTr.sorted.STA...                                           1.000000                                               \n",
       "inputs/se_single/ENCFF201PEI.fastqTr.sorted.STA...                                           0.983265                                               \n",
       "\n",
       "                                                    inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           0.972402                                               \n",
       "inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           0.984047                                               \n",
       "inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STA...                                           0.972402                                               \n",
       "inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STA...                                           0.984047                                               \n",
       "inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           0.983265                                               \n",
       "inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           1.000000                                               \n",
       "inputs/se_single/ENCFF040CTS.fastqTr.sorted.STA...                                           0.983265                                               \n",
       "inputs/se_single/ENCFF201PEI.fastqTr.sorted.STA...                                           1.000000                                               \n",
       "\n",
       "                                                    inputs/se_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           0.992351                                                \n",
       "inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           0.968103                                                \n",
       "inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STA...                                           0.992351                                                \n",
       "inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STA...                                           0.968103                                                \n",
       "inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           1.000000                                                \n",
       "inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           0.983265                                                \n",
       "inputs/se_single/ENCFF040CTS.fastqTr.sorted.STA...                                           1.000000                                                \n",
       "inputs/se_single/ENCFF201PEI.fastqTr.sorted.STA...                                           0.983265                                                \n",
       "\n",
       "                                                    inputs/se_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \n",
       "inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           0.972402                                               \n",
       "inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           0.984047                                               \n",
       "inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STA...                                           0.972402                                               \n",
       "inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STA...                                           0.984047                                               \n",
       "inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STAR...                                           0.983265                                               \n",
       "inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STAR...                                           1.000000                                               \n",
       "inputs/se_single/ENCFF040CTS.fastqTr.sorted.STA...                                           0.983265                                               \n",
       "inputs/se_single/ENCFF201PEI.fastqTr.sorted.STA...                                           1.000000                                               "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "featurecounts_output.corr()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Now check the starLog final outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 11:22:00\n",
      "                             Started mapping on |\tAug 05 11:22:27\n",
      "                                    Finished on |\tAug 05 12:32:35\n",
      "       Mapping speed, Million of reads per hour |\t34.28\n",
      "\n",
      "                          Number of input reads |\t40071659\n",
      "                      Average input read length |\t198\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t151067\n",
      "                        Uniquely mapped reads % |\t0.38%\n",
      "                          Average mapped length |\t200.65\n",
      "                       Number of splices: Total |\t316\n",
      "            Number of splices: Annotated (sjdb) |\t0\n",
      "                       Number of splices: GT/AG |\t217\n",
      "                       Number of splices: GC/AG |\t6\n",
      "                       Number of splices: AT/AC |\t1\n",
      "               Number of splices: Non-canonical |\t92\n",
      "                      Mismatch rate per base, % |\t1.62%\n",
      "                         Deletion rate per base |\t0.10%\n",
      "                        Deletion average length |\t2.60\n",
      "                        Insertion rate per base |\t0.02%\n",
      "                       Insertion average length |\t1.69\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t129372\n",
      "             % of reads mapped to multiple loci |\t0.32%\n",
      "        Number of reads mapped to too many loci |\t540\n",
      "             % of reads mapped to too many loci |\t0.00%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t0.55%\n",
      "                 % of reads unmapped: too short |\t6.13%\n",
      "                     % of reads unmapped: other |\t92.62%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 19:25:13\n",
      "                             Started mapping on |\tAug 05 19:25:39\n",
      "                                    Finished on |\tAug 05 20:38:06\n",
      "       Mapping speed, Million of reads per hour |\t33.19\n",
      "\n",
      "                          Number of input reads |\t40071659\n",
      "                      Average input read length |\t198\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t151067\n",
      "                        Uniquely mapped reads % |\t0.38%\n",
      "                          Average mapped length |\t200.65\n",
      "                       Number of splices: Total |\t316\n",
      "            Number of splices: Annotated (sjdb) |\t0\n",
      "                       Number of splices: GT/AG |\t217\n",
      "                       Number of splices: GC/AG |\t6\n",
      "                       Number of splices: AT/AC |\t1\n",
      "               Number of splices: Non-canonical |\t92\n",
      "                      Mismatch rate per base, % |\t1.62%\n",
      "                         Deletion rate per base |\t0.10%\n",
      "                        Deletion average length |\t2.60\n",
      "                        Insertion rate per base |\t0.02%\n",
      "                       Insertion average length |\t1.69\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t129372\n",
      "             % of reads mapped to multiple loci |\t0.32%\n",
      "        Number of reads mapped to too many loci |\t540\n",
      "             % of reads mapped to too many loci |\t0.00%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t0.55%\n",
      "                 % of reads unmapped: too short |\t6.13%\n",
      "                     % of reads unmapped: other |\t92.62%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/pe_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 13:02:52\n",
      "                             Started mapping on |\tAug 05 13:03:17\n",
      "                                    Finished on |\tAug 05 13:20:26\n",
      "       Mapping speed, Million of reads per hour |\t139.21\n",
      "\n",
      "                          Number of input reads |\t39791541\n",
      "                      Average input read length |\t198\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t30503888\n",
      "                        Uniquely mapped reads % |\t76.66%\n",
      "                          Average mapped length |\t200.75\n",
      "                       Number of splices: Total |\t17548728\n",
      "            Number of splices: Annotated (sjdb) |\t17423638\n",
      "                       Number of splices: GT/AG |\t17363709\n",
      "                       Number of splices: GC/AG |\t149696\n",
      "                       Number of splices: AT/AC |\t18972\n",
      "               Number of splices: Non-canonical |\t16351\n",
      "                      Mismatch rate per base, % |\t0.36%\n",
      "                         Deletion rate per base |\t0.03%\n",
      "                        Deletion average length |\t2.50\n",
      "                        Insertion rate per base |\t0.00%\n",
      "                       Insertion average length |\t1.74\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t1062259\n",
      "             % of reads mapped to multiple loci |\t2.67%\n",
      "        Number of reads mapped to too many loci |\t9490\n",
      "             % of reads mapped to too many loci |\t0.02%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t1.56%\n",
      "                 % of reads unmapped: too short |\t18.81%\n",
      "                     % of reads unmapped: other |\t0.27%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/pe_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 20:50:00\n",
      "                             Started mapping on |\tAug 05 20:50:34\n",
      "                                    Finished on |\tAug 05 20:59:56\n",
      "       Mapping speed, Million of reads per hour |\t254.89\n",
      "\n",
      "                          Number of input reads |\t39791541\n",
      "                      Average input read length |\t198\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t30503888\n",
      "                        Uniquely mapped reads % |\t76.66%\n",
      "                          Average mapped length |\t200.75\n",
      "                       Number of splices: Total |\t17548728\n",
      "            Number of splices: Annotated (sjdb) |\t17423638\n",
      "                       Number of splices: GT/AG |\t17363709\n",
      "                       Number of splices: GC/AG |\t149696\n",
      "                       Number of splices: AT/AC |\t18972\n",
      "               Number of splices: Non-canonical |\t16351\n",
      "                      Mismatch rate per base, % |\t0.36%\n",
      "                         Deletion rate per base |\t0.03%\n",
      "                        Deletion average length |\t2.50\n",
      "                        Insertion rate per base |\t0.00%\n",
      "                       Insertion average length |\t1.74\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t1062259\n",
      "             % of reads mapped to multiple loci |\t2.67%\n",
      "        Number of reads mapped to too many loci |\t9490\n",
      "             % of reads mapped to too many loci |\t0.02%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t1.56%\n",
      "                 % of reads unmapped: too short |\t18.81%\n",
      "                     % of reads unmapped: other |\t0.27%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 11:33:52\n",
      "                             Started mapping on |\tAug 05 11:34:32\n",
      "                                    Finished on |\tAug 05 12:54:12\n",
      "       Mapping speed, Million of reads per hour |\t34.91\n",
      "\n",
      "                          Number of input reads |\t46358745\n",
      "                      Average input read length |\t197\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t199635\n",
      "                        Uniquely mapped reads % |\t0.43%\n",
      "                          Average mapped length |\t200.53\n",
      "                       Number of splices: Total |\t287\n",
      "            Number of splices: Annotated (sjdb) |\t0\n",
      "                       Number of splices: GT/AG |\t200\n",
      "                       Number of splices: GC/AG |\t6\n",
      "                       Number of splices: AT/AC |\t0\n",
      "               Number of splices: Non-canonical |\t81\n",
      "                      Mismatch rate per base, % |\t1.26%\n",
      "                         Deletion rate per base |\t0.07%\n",
      "                        Deletion average length |\t2.87\n",
      "                        Insertion rate per base |\t0.02%\n",
      "                       Insertion average length |\t1.68\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t262499\n",
      "             % of reads mapped to multiple loci |\t0.57%\n",
      "        Number of reads mapped to too many loci |\t595\n",
      "             % of reads mapped to too many loci |\t0.00%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t0.44%\n",
      "                 % of reads unmapped: too short |\t6.36%\n",
      "                     % of reads unmapped: other |\t92.20%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 14:11:19\n",
      "                             Started mapping on |\tAug 05 14:12:15\n",
      "                                    Finished on |\tAug 05 15:37:38\n",
      "       Mapping speed, Million of reads per hour |\t32.58\n",
      "\n",
      "                          Number of input reads |\t46358745\n",
      "                      Average input read length |\t197\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t199635\n",
      "                        Uniquely mapped reads % |\t0.43%\n",
      "                          Average mapped length |\t200.53\n",
      "                       Number of splices: Total |\t287\n",
      "            Number of splices: Annotated (sjdb) |\t0\n",
      "                       Number of splices: GT/AG |\t200\n",
      "                       Number of splices: GC/AG |\t6\n",
      "                       Number of splices: AT/AC |\t0\n",
      "               Number of splices: Non-canonical |\t81\n",
      "                      Mismatch rate per base, % |\t1.26%\n",
      "                         Deletion rate per base |\t0.07%\n",
      "                        Deletion average length |\t2.87\n",
      "                        Insertion rate per base |\t0.02%\n",
      "                       Insertion average length |\t1.68\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t262499\n",
      "             % of reads mapped to multiple loci |\t0.57%\n",
      "        Number of reads mapped to too many loci |\t595\n",
      "             % of reads mapped to too many loci |\t0.00%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t0.44%\n",
      "                 % of reads unmapped: too short |\t6.36%\n",
      "                     % of reads unmapped: other |\t92.20%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/pe_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 13:36:31\n",
      "                             Started mapping on |\tAug 05 13:36:49\n",
      "                                    Finished on |\tAug 05 13:54:52\n",
      "       Mapping speed, Million of reads per hour |\t152.57\n",
      "\n",
      "                          Number of input reads |\t45897169\n",
      "                      Average input read length |\t197\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t32584453\n",
      "                        Uniquely mapped reads % |\t70.99%\n",
      "                          Average mapped length |\t200.67\n",
      "                       Number of splices: Total |\t21491314\n",
      "            Number of splices: Annotated (sjdb) |\t21344145\n",
      "                       Number of splices: GT/AG |\t21258976\n",
      "                       Number of splices: GC/AG |\t187811\n",
      "                       Number of splices: AT/AC |\t24220\n",
      "               Number of splices: Non-canonical |\t20307\n",
      "                      Mismatch rate per base, % |\t0.36%\n",
      "                         Deletion rate per base |\t0.03%\n",
      "                        Deletion average length |\t2.58\n",
      "                        Insertion rate per base |\t0.00%\n",
      "                       Insertion average length |\t1.70\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t1212778\n",
      "             % of reads mapped to multiple loci |\t2.64%\n",
      "        Number of reads mapped to too many loci |\t11974\n",
      "             % of reads mapped to too many loci |\t0.03%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t1.70%\n",
      "                 % of reads unmapped: too short |\t23.97%\n",
      "                     % of reads unmapped: other |\t0.67%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/pe_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 15:46:27\n",
      "                             Started mapping on |\tAug 05 15:46:55\n",
      "                                    Finished on |\tAug 05 15:59:23\n",
      "       Mapping speed, Million of reads per hour |\t220.90\n",
      "\n",
      "                          Number of input reads |\t45897169\n",
      "                      Average input read length |\t197\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t32584453\n",
      "                        Uniquely mapped reads % |\t70.99%\n",
      "                          Average mapped length |\t200.67\n",
      "                       Number of splices: Total |\t21491314\n",
      "            Number of splices: Annotated (sjdb) |\t21344145\n",
      "                       Number of splices: GT/AG |\t21258976\n",
      "                       Number of splices: GC/AG |\t187811\n",
      "                       Number of splices: AT/AC |\t24220\n",
      "               Number of splices: Non-canonical |\t20307\n",
      "                      Mismatch rate per base, % |\t0.36%\n",
      "                         Deletion rate per base |\t0.03%\n",
      "                        Deletion average length |\t2.58\n",
      "                        Insertion rate per base |\t0.00%\n",
      "                       Insertion average length |\t1.70\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t1212778\n",
      "             % of reads mapped to multiple loci |\t2.64%\n",
      "        Number of reads mapped to too many loci |\t11974\n",
      "             % of reads mapped to too many loci |\t0.03%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t1.70%\n",
      "                 % of reads unmapped: too short |\t23.97%\n",
      "                     % of reads unmapped: other |\t0.67%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n"
     ]
    }
   ],
   "source": [
    "for s in ['ENCFF040CTS', 'ENCFF201PEI']: # for each rep\n",
    "    for c in ['.fastqTr.sorted.STARLog.final.out', '.fastqTr.sorted.STARUnmapped.out.sorted.STARLog.final.out']: # for repeat-mapped, genome-mapped\n",
    "        for pe_log in sorted(glob.glob(os.path.join(input_dir, 'pe_*/{}{}'.format(s, c)))):\n",
    "            print(pe_log)\n",
    "            ! cat $pe_log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 10:58:16\n",
      "                             Started mapping on |\tAug 05 10:58:51\n",
      "                                    Finished on |\tAug 05 11:19:40\n",
      "       Mapping speed, Million of reads per hour |\t115.50\n",
      "\n",
      "                          Number of input reads |\t40071659\n",
      "                      Average input read length |\t97\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t330591\n",
      "                        Uniquely mapped reads % |\t0.82%\n",
      "                          Average mapped length |\t96.33\n",
      "                       Number of splices: Total |\t848\n",
      "            Number of splices: Annotated (sjdb) |\t0\n",
      "                       Number of splices: GT/AG |\t747\n",
      "                       Number of splices: GC/AG |\t14\n",
      "                       Number of splices: AT/AC |\t1\n",
      "               Number of splices: Non-canonical |\t86\n",
      "                      Mismatch rate per base, % |\t4.29%\n",
      "                         Deletion rate per base |\t0.21%\n",
      "                        Deletion average length |\t2.10\n",
      "                        Insertion rate per base |\t0.06%\n",
      "                       Insertion average length |\t1.56\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t241489\n",
      "             % of reads mapped to multiple loci |\t0.60%\n",
      "        Number of reads mapped to too many loci |\t5940\n",
      "             % of reads mapped to too many loci |\t0.01%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t0.37%\n",
      "                 % of reads unmapped: too short |\t3.95%\n",
      "                     % of reads unmapped: other |\t94.23%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/se_single/ENCFF040CTS.fastqTr.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 20:52:08\n",
      "                             Started mapping on |\tAug 05 20:52:36\n",
      "                                    Finished on |\tAug 05 21:11:47\n",
      "       Mapping speed, Million of reads per hour |\t125.33\n",
      "\n",
      "                          Number of input reads |\t40071659\n",
      "                      Average input read length |\t97\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t330591\n",
      "                        Uniquely mapped reads % |\t0.82%\n",
      "                          Average mapped length |\t96.33\n",
      "                       Number of splices: Total |\t848\n",
      "            Number of splices: Annotated (sjdb) |\t0\n",
      "                       Number of splices: GT/AG |\t747\n",
      "                       Number of splices: GC/AG |\t14\n",
      "                       Number of splices: AT/AC |\t1\n",
      "               Number of splices: Non-canonical |\t86\n",
      "                      Mismatch rate per base, % |\t4.29%\n",
      "                         Deletion rate per base |\t0.21%\n",
      "                        Deletion average length |\t2.10\n",
      "                        Insertion rate per base |\t0.06%\n",
      "                       Insertion average length |\t1.56\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t241489\n",
      "             % of reads mapped to multiple loci |\t0.60%\n",
      "        Number of reads mapped to too many loci |\t5940\n",
      "             % of reads mapped to too many loci |\t0.01%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t0.37%\n",
      "                 % of reads unmapped: too short |\t3.95%\n",
      "                     % of reads unmapped: other |\t94.23%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/se_multi/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 11:47:04\n",
      "                             Started mapping on |\tAug 05 11:47:53\n",
      "                                    Finished on |\tAug 05 11:54:00\n",
      "       Mapping speed, Million of reads per hour |\t387.46\n",
      "\n",
      "                          Number of input reads |\t39499579\n",
      "                      Average input read length |\t97\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t36836098\n",
      "                        Uniquely mapped reads % |\t93.26%\n",
      "                          Average mapped length |\t97.55\n",
      "                       Number of splices: Total |\t10068609\n",
      "            Number of splices: Annotated (sjdb) |\t9998257\n",
      "                       Number of splices: GT/AG |\t9963453\n",
      "                       Number of splices: GC/AG |\t81288\n",
      "                       Number of splices: AT/AC |\t11596\n",
      "               Number of splices: Non-canonical |\t12272\n",
      "                      Mismatch rate per base, % |\t0.53%\n",
      "                         Deletion rate per base |\t0.04%\n",
      "                        Deletion average length |\t2.49\n",
      "                        Insertion rate per base |\t0.00%\n",
      "                       Insertion average length |\t1.67\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t2063747\n",
      "             % of reads mapped to multiple loci |\t5.22%\n",
      "        Number of reads mapped to too many loci |\t11630\n",
      "             % of reads mapped to too many loci |\t0.03%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t0.49%\n",
      "                 % of reads unmapped: too short |\t0.73%\n",
      "                     % of reads unmapped: other |\t0.28%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/se_single/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 21:35:50\n",
      "                             Started mapping on |\tAug 05 21:36:23\n",
      "                                    Finished on |\tAug 05 21:40:40\n",
      "       Mapping speed, Million of reads per hour |\t553.30\n",
      "\n",
      "                          Number of input reads |\t39499579\n",
      "                      Average input read length |\t97\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t36836098\n",
      "                        Uniquely mapped reads % |\t93.26%\n",
      "                          Average mapped length |\t97.55\n",
      "                       Number of splices: Total |\t10068609\n",
      "            Number of splices: Annotated (sjdb) |\t9998257\n",
      "                       Number of splices: GT/AG |\t9963453\n",
      "                       Number of splices: GC/AG |\t81288\n",
      "                       Number of splices: AT/AC |\t11596\n",
      "               Number of splices: Non-canonical |\t12272\n",
      "                      Mismatch rate per base, % |\t0.53%\n",
      "                         Deletion rate per base |\t0.04%\n",
      "                        Deletion average length |\t2.49\n",
      "                        Insertion rate per base |\t0.00%\n",
      "                       Insertion average length |\t1.67\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t2063747\n",
      "             % of reads mapped to multiple loci |\t5.22%\n",
      "        Number of reads mapped to too many loci |\t11630\n",
      "             % of reads mapped to too many loci |\t0.03%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t0.49%\n",
      "                 % of reads unmapped: too short |\t0.73%\n",
      "                     % of reads unmapped: other |\t0.28%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 11:13:27\n",
      "                             Started mapping on |\tAug 05 11:13:56\n",
      "                                    Finished on |\tAug 05 11:38:05\n",
      "       Mapping speed, Million of reads per hour |\t115.18\n",
      "\n",
      "                          Number of input reads |\t46358745\n",
      "                      Average input read length |\t96\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t401475\n",
      "                        Uniquely mapped reads % |\t0.87%\n",
      "                          Average mapped length |\t94.99\n",
      "                       Number of splices: Total |\t861\n",
      "            Number of splices: Annotated (sjdb) |\t0\n",
      "                       Number of splices: GT/AG |\t688\n",
      "                       Number of splices: GC/AG |\t27\n",
      "                       Number of splices: AT/AC |\t2\n",
      "               Number of splices: Non-canonical |\t144\n",
      "                      Mismatch rate per base, % |\t3.57%\n",
      "                         Deletion rate per base |\t0.16%\n",
      "                        Deletion average length |\t2.17\n",
      "                        Insertion rate per base |\t0.06%\n",
      "                       Insertion average length |\t1.47\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t437060\n",
      "             % of reads mapped to multiple loci |\t0.94%\n",
      "        Number of reads mapped to too many loci |\t7417\n",
      "             % of reads mapped to too many loci |\t0.02%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t0.32%\n",
      "                 % of reads unmapped: too short |\t4.20%\n",
      "                     % of reads unmapped: other |\t93.66%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/se_single/ENCFF201PEI.fastqTr.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 17:52:02\n",
      "                             Started mapping on |\tAug 05 17:52:25\n",
      "                                    Finished on |\tAug 05 18:17:38\n",
      "       Mapping speed, Million of reads per hour |\t110.31\n",
      "\n",
      "                          Number of input reads |\t46358745\n",
      "                      Average input read length |\t96\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t401475\n",
      "                        Uniquely mapped reads % |\t0.87%\n",
      "                          Average mapped length |\t94.99\n",
      "                       Number of splices: Total |\t861\n",
      "            Number of splices: Annotated (sjdb) |\t0\n",
      "                       Number of splices: GT/AG |\t688\n",
      "                       Number of splices: GC/AG |\t27\n",
      "                       Number of splices: AT/AC |\t2\n",
      "               Number of splices: Non-canonical |\t144\n",
      "                      Mismatch rate per base, % |\t3.57%\n",
      "                         Deletion rate per base |\t0.16%\n",
      "                        Deletion average length |\t2.17\n",
      "                        Insertion rate per base |\t0.06%\n",
      "                       Insertion average length |\t1.47\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t437060\n",
      "             % of reads mapped to multiple loci |\t0.94%\n",
      "        Number of reads mapped to too many loci |\t7417\n",
      "             % of reads mapped to too many loci |\t0.02%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t0.32%\n",
      "                 % of reads unmapped: too short |\t4.20%\n",
      "                     % of reads unmapped: other |\t93.66%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/se_multi/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 12:07:00\n",
      "                             Started mapping on |\tAug 05 12:07:19\n",
      "                                    Finished on |\tAug 05 12:12:47\n",
      "       Mapping speed, Million of reads per hour |\t499.61\n",
      "\n",
      "                          Number of input reads |\t45520210\n",
      "                      Average input read length |\t96\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t42056939\n",
      "                        Uniquely mapped reads % |\t92.39%\n",
      "                          Average mapped length |\t96.72\n",
      "                       Number of splices: Total |\t13210511\n",
      "            Number of splices: Annotated (sjdb) |\t13122873\n",
      "                       Number of splices: GT/AG |\t13070085\n",
      "                       Number of splices: GC/AG |\t109098\n",
      "                       Number of splices: AT/AC |\t16079\n",
      "               Number of splices: Non-canonical |\t15249\n",
      "                      Mismatch rate per base, % |\t0.59%\n",
      "                         Deletion rate per base |\t0.03%\n",
      "                        Deletion average length |\t2.53\n",
      "                        Insertion rate per base |\t0.00%\n",
      "                       Insertion average length |\t1.63\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t2452594\n",
      "             % of reads mapped to multiple loci |\t5.39%\n",
      "        Number of reads mapped to too many loci |\t15608\n",
      "             % of reads mapped to too many loci |\t0.03%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t0.52%\n",
      "                 % of reads unmapped: too short |\t0.97%\n",
      "                     % of reads unmapped: other |\t0.69%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n",
      "/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_single_vs_multinode/inputs/se_single/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARLog.final.out\n",
      "                                 Started job on |\tAug 05 18:56:24\n",
      "                             Started mapping on |\tAug 05 18:57:24\n",
      "                                    Finished on |\tAug 05 19:06:09\n",
      "       Mapping speed, Million of reads per hour |\t312.14\n",
      "\n",
      "                          Number of input reads |\t45520210\n",
      "                      Average input read length |\t96\n",
      "                                    UNIQUE READS:\n",
      "                   Uniquely mapped reads number |\t42056939\n",
      "                        Uniquely mapped reads % |\t92.39%\n",
      "                          Average mapped length |\t96.72\n",
      "                       Number of splices: Total |\t13210511\n",
      "            Number of splices: Annotated (sjdb) |\t13122873\n",
      "                       Number of splices: GT/AG |\t13070085\n",
      "                       Number of splices: GC/AG |\t109098\n",
      "                       Number of splices: AT/AC |\t16079\n",
      "               Number of splices: Non-canonical |\t15249\n",
      "                      Mismatch rate per base, % |\t0.59%\n",
      "                         Deletion rate per base |\t0.03%\n",
      "                        Deletion average length |\t2.53\n",
      "                        Insertion rate per base |\t0.00%\n",
      "                       Insertion average length |\t1.63\n",
      "                             MULTI-MAPPING READS:\n",
      "        Number of reads mapped to multiple loci |\t2452594\n",
      "             % of reads mapped to multiple loci |\t5.39%\n",
      "        Number of reads mapped to too many loci |\t15608\n",
      "             % of reads mapped to too many loci |\t0.03%\n",
      "                                  UNMAPPED READS:\n",
      "       % of reads unmapped: too many mismatches |\t0.52%\n",
      "                 % of reads unmapped: too short |\t0.97%\n",
      "                     % of reads unmapped: other |\t0.69%\n",
      "                                  CHIMERIC READS:\n",
      "                       Number of chimeric reads |\t0\n",
      "                            % of chimeric reads |\t0.00%\n"
     ]
    }
   ],
   "source": [
    "for s in ['ENCFF040CTS', 'ENCFF201PEI']: # for each rep\n",
    "    for c in ['.fastqTr.sorted.STARLog.final.out', '.fastqTr.sorted.STARUnmapped.out.sorted.STARLog.final.out']: # for repeat-mapped, genome-mapped\n",
    "        for se_log in sorted(glob.glob(os.path.join(input_dir, 'se_*/{}{}'.format(s, c)))):\n",
    "            print(se_log)\n",
    "            ! cat $se_log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python3-essential-tscc",
   "language": "python",
   "name": "python3essentialtscc"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
