{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This notebook compares the outputs from our basic RNAseq pipeline and the \"official ENCODE\" version. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "input_dir = '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_pipeline_to_encode/inputs/'\n",
    "output_dir = '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_pipeline_to_encode/outputs/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_pipeline_to_encode/inputs/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam',\n",
       " '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_pipeline_to_encode/inputs/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam',\n",
       " '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_pipeline_to_encode/inputs/ENCFF511KFD.bam',\n",
       " '/home/bay001/projects/codebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_pipeline_to_encode/inputs/ENCFF880HKN.bam']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_bams = sorted(glob.glob(os.path.join(input_dir, '*.bam')))\n",
    "all_bams"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v1.5.3\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m4 BAM files  \u001b[0m \u001b[0m                                   ||\n",
      "||                           \u001b[32mP\u001b[36m inputs/ENCFF040CTS.fastqTr.sorted.STARUnma\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mP\u001b[36m inputs/ENCFF201PEI.fastqTr.sorted.STARUnma\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mP\u001b[36m inputs/ENCFF511KFD.bam\u001b[0m \u001b[0m                        ||\n",
      "||                           \u001b[32mP\u001b[36m inputs/ENCFF880HKN.bam\u001b[0m \u001b[0m                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36m/home/bay001/projects/codebase/bulk_rnaseq_p\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||                 Summary : \u001b[36m/home/bay001/projects/codebase/bulk_rnaseq_p\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||              Annotation : \u001b[36m/projects/ps-yeolab3/bay001/annotations/hg19\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||      Dir for temp files : \u001b[36m/home/bay001/projects/codebase/bulk_rnaseq_p\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||         Strand specific : \u001b[36mreversely stranded\u001b[0m \u001b[0m                              ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\=====================\u001b[36m http://subread.sourceforge.net/ \u001b[0m======================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file /projects/ps-yeolab3/bay001/annotations/hg19/genc\u001b[0m ... \u001b[0m||\n",
      "||    Features : \u001b[36m1196293\u001b[0m \u001b[0m                                                     ||\n",
      "||    Meta-features : \u001b[36m57820\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m25\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file inputs/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.so\u001b[0m ... \u001b[0m||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Assign reads to features... \u001b[0m                                            ||\n",
      "||    Total reads : \u001b[36m83504246\u001b[0m \u001b[0m                                                 ||\n",
      "||    Successfully assigned reads : \u001b[36m27263390 (32.6%)\u001b[0m \u001b[0m                         ||\n",
      "||    Running time : \u001b[36m2.41 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file inputs/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.so\u001b[0m ... \u001b[0m||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Assign reads to features... \u001b[0m                                            ||\n",
      "||    Total reads : \u001b[36m96531504\u001b[0m \u001b[0m                                                 ||\n",
      "||    Successfully assigned reads : \u001b[36m29563998 (30.6%)\u001b[0m \u001b[0m                         ||\n",
      "||    Running time : \u001b[36m2.79 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file inputs/ENCFF511KFD.bam... \u001b[0m                                ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Assign reads to features... \u001b[0m                                            ||\n",
      "||    Total reads : \u001b[36m85594236\u001b[0m \u001b[0m                                                 ||\n",
      "||    Successfully assigned reads : \u001b[36m30462844 (35.6%)\u001b[0m \u001b[0m                         ||\n",
      "||    Running time : \u001b[36m5.39 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file inputs/ENCFF880HKN.bam... \u001b[0m                                ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Assign reads to features... \u001b[0m                                            ||\n",
      "||    Total reads : \u001b[36m99864052\u001b[0m \u001b[0m                                                 ||\n",
      "||    Successfully assigned reads : \u001b[36m34293828 (34.3%)\u001b[0m \u001b[0m                         ||\n",
      "||    Running time : \u001b[36m6.26 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                         \u001b[36mRead assignment finished.\u001b[0m \u001b[0m                         ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"/home/bay001/projects/c \u001b[0m ||\n",
      "|| odebase/bulk_rnaseq_pipeline/tests/0.0.4/compare_pipeline_to_encode/outpu \u001b[0m ||\n",
      "|| ts/counts.txt\" \u001b[0m                                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\=====================\u001b[36m http://subread.sourceforge.net/ \u001b[0m======================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "gencode = '/projects/ps-yeolab3/bay001/annotations/hg19/gencode_v19/gencode.v19.annotation.gtf'\n",
    "counts = os.path.join(output_dir, 'counts.txt')\n",
    "\n",
    "cmd = 'module load subreadfeaturecounts;featureCounts '\n",
    "cmd += '-a {} -s 2 -o {} inputs/*.bam'.format(gencode, counts)\n",
    "\n",
    "! $cmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>inputs/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/ENCFF511KFD.bam</th>\n",
       "      <th>inputs/ENCFF880HKN.bam</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Geneid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000223972.4</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000227232.4</th>\n",
       "      <td>52</td>\n",
       "      <td>118</td>\n",
       "      <td>59</td>\n",
       "      <td>132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000243485.2</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000237613.2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000268020.2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   inputs/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "Geneid                                                                                                    \n",
       "ENSG00000223972.4                                                  1                                      \n",
       "ENSG00000227232.4                                                 52                                      \n",
       "ENSG00000243485.2                                                  1                                      \n",
       "ENSG00000237613.2                                                  0                                      \n",
       "ENSG00000268020.2                                                  0                                      \n",
       "\n",
       "                   inputs/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "Geneid                                                                                                    \n",
       "ENSG00000223972.4                                                  0                                      \n",
       "ENSG00000227232.4                                                118                                      \n",
       "ENSG00000243485.2                                                  0                                      \n",
       "ENSG00000237613.2                                                  0                                      \n",
       "ENSG00000268020.2                                                  0                                      \n",
       "\n",
       "                   inputs/ENCFF511KFD.bam  inputs/ENCFF880HKN.bam  \n",
       "Geneid                                                             \n",
       "ENSG00000223972.4                       1                       0  \n",
       "ENSG00000227232.4                      59                     132  \n",
       "ENSG00000243485.2                       1                       0  \n",
       "ENSG00000237613.2                       0                       0  \n",
       "ENSG00000268020.2                       0                       0  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "featurecounts_output = pd.read_csv(counts, sep='\\t', skiprows=1, index_col=0)\n",
    "featurecounts_output = featurecounts_output.iloc[:,5:]\n",
    "featurecounts_output.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>inputs/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <th>inputs/ENCFF511KFD.bam</th>\n",
       "      <th>inputs/ENCFF880HKN.bam</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>inputs/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.977703</td>\n",
       "      <td>0.995985</td>\n",
       "      <td>0.972401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>inputs/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam</th>\n",
       "      <td>0.977703</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.978242</td>\n",
       "      <td>0.996011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>inputs/ENCFF511KFD.bam</th>\n",
       "      <td>0.995985</td>\n",
       "      <td>0.978242</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.980722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>inputs/ENCFF880HKN.bam</th>\n",
       "      <td>0.972401</td>\n",
       "      <td>0.996011</td>\n",
       "      <td>0.980722</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    inputs/ENCFF040CTS.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "inputs/ENCFF040CTS.fastqTr.sorted.STARUnmapped....                                           1.000000                                      \n",
       "inputs/ENCFF201PEI.fastqTr.sorted.STARUnmapped....                                           0.977703                                      \n",
       "inputs/ENCFF511KFD.bam                                                                       0.995985                                      \n",
       "inputs/ENCFF880HKN.bam                                                                       0.972401                                      \n",
       "\n",
       "                                                    inputs/ENCFF201PEI.fastqTr.sorted.STARUnmapped.out.sorted.STARAligned.out.sorted.bam  \\\n",
       "inputs/ENCFF040CTS.fastqTr.sorted.STARUnmapped....                                           0.977703                                      \n",
       "inputs/ENCFF201PEI.fastqTr.sorted.STARUnmapped....                                           1.000000                                      \n",
       "inputs/ENCFF511KFD.bam                                                                       0.978242                                      \n",
       "inputs/ENCFF880HKN.bam                                                                       0.996011                                      \n",
       "\n",
       "                                                    inputs/ENCFF511KFD.bam  \\\n",
       "inputs/ENCFF040CTS.fastqTr.sorted.STARUnmapped....                0.995985   \n",
       "inputs/ENCFF201PEI.fastqTr.sorted.STARUnmapped....                0.978242   \n",
       "inputs/ENCFF511KFD.bam                                            1.000000   \n",
       "inputs/ENCFF880HKN.bam                                            0.980722   \n",
       "\n",
       "                                                    inputs/ENCFF880HKN.bam  \n",
       "inputs/ENCFF040CTS.fastqTr.sorted.STARUnmapped....                0.972401  \n",
       "inputs/ENCFF201PEI.fastqTr.sorted.STARUnmapped....                0.996011  \n",
       "inputs/ENCFF511KFD.bam                                            0.980722  \n",
       "inputs/ENCFF880HKN.bam                                            1.000000  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "featurecounts_output.corr()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python3-essential-tscc",
   "language": "python",
   "name": "python3essentialtscc"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
