- fastqc '/projects/ps-yeolab/seqdata/20160606_stefan_parp13/SAMPLE.fastq.gz'  '-o' '/home/bay001/projects/parp13_ago2_20160606/analysis/parp13_ago2_v3'  > /DIRECTORY/SAMPLE.dummy_fastqc 
- cutadapt -f fastq '--match-read-wildcards'  '--times' '2'  '-e' '0.0'  '-O' '5'  '--quality-cutoff' '6'  '-m' '18'  '-b' 'TCGTATGCCGTCTTCTGCTTG' '-b' 'ATCTCGTATGCCGTCTTCTGCTTG' '-b' 'CGACAGGTTCAGAGTTCTACAGTCCGACGATC' '-b' 'GATCGGAAGAGCACACGTCTGAACTCCAGTCAC' '-b' 'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA' '-b' 'TTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT'  '-o' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.fastq'  '/projects/ps-yeolab/seqdata/20160606_stefan_parp13/SAMPLE.fastq.gz'  > SAMPLE.polyATrim.adapterTrim.metrics 
- STAR  '--runMode' 'alignReads'  '--runThreadN' '16'  '--genomeDir' '/projects/ps-yeolab/genomes/RepBase18.05.fasta/STAR_fixed'  '--genomeLoad' 'LoadAndRemove'  '--readFilesIn' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.fastq'  '--outSAMunmapped' 'Within'  '--outFilterMultimapNmax' '10'  '--outFilterMultimapScoreRange' '1'  '--outFileNamePrefix' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rep.bam'  '--outSAMattributes' 'All'  '--outSAMstrandField intronMotif'  '--outStd' 'BAM_Unsorted'  '--outSAMtype' 'BAM' 'Unsorted'  '--outFilterType' 'BySJout'  '--outReadsUnmapped' 'Fastx'  '--outFilterScoreMin' '10'  '--outSAMattrRGline' 'ID:foo' > /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rep.bam 
- samtools view /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rep.bam | count_aligned_from_sam.py > /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.metrics 
- fastqc '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rep.bamUnmapped.out.mate1'  '-o' '/home/bay001/projects/parp13_ago2_20160606/analysis/parp13_ago2_v3'  > /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rep.bamUnmapped.out.mate1.dummy_fastqc 
- STAR  '--runMode' 'alignReads'  '--runThreadN' '16'  '--genomeDir' '/projects/ps-yeolab/genomes/hg19/star_sjdb'  '--genomeLoad' 'LoadAndRemove'  '--readFilesIn' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rep.bamUnmapped.out.mate1'  '--outSAMunmapped' 'Within'  '--outFilterMultimapNmax' '10'  '--outFilterMultimapScoreRange' '1'  '--outFileNamePrefix' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.bam'  '--outSAMattributes' 'All'  '--outSAMstrandField intronMotif'  '--outStd' 'BAM_Unsorted'  '--outSAMtype' 'BAM' 'Unsorted'  '--outFilterType' 'BySJout'  '--outReadsUnmapped' 'Fastx'  '--outFilterScoreMin' '10'  '--outSAMattrRGline' 'ID:foo' > /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.bam 
-  'java'  '-Xmx2048m'  '-XX:+UseParallelOldGC'  '-XX:ParallelGCThreads=4'  '-XX:GCTimeLimit=50'  '-XX:GCHeapFreeLimit=10'  '-Djava.io.tmpdir=/DIRECTORY/.queue/tmp'  '-cp' '/projects/ps-yeolab/software/gatk/dist/Queue.jar'  'net.sf.picard.sam.SortSam'  'INPUT=/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.bam'  'TMP_DIR=/DIRECTORY/.queue/tmp'  'OUTPUT=/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.bam'  'VALIDATION_STRINGENCY=SILENT'  'SO=coordinate'  'CREATE_INDEX=true'  
-  'java'  '-Xmx2048m'  '-XX:+UseParallelOldGC'  '-XX:ParallelGCThreads=4'  '-XX:GCTimeLimit=50'  '-XX:GCHeapFreeLimit=10'  '-Djava.io.tmpdir=/DIRECTORY/.queue/tmp'  '-cp' '/projects/ps-yeolab/software/gatk/dist/Queue.jar'  'net.sf.picard.sam.AddOrReplaceReadGroups'  'INPUT=/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.bam'  'TMP_DIR=/DIRECTORY/.queue/tmp'  'OUTPUT=/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.bam'  'VALIDATION_STRINGENCY=SILENT'  'SO=coordinate'  'CREATE_INDEX=true'  'RGID=foo'  'RGLB=foo'  'RGPL=illumina'  'RGPU=bar'  'RGSM=baz'  'RGCN=Biogem'  
-  'samtools'  'index'  '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.bam'  '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.bam.bai'  
-  samtools view -h /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.bam | awk 'BEGIN {OFS="\t"} {if(!!and($2,0x0080)) {if(!!and($2, 0x0004)) {$2 = $2 - 16} else {$2 = $2 + 16}}; print $0}'  | samtools view -bS - | genomeCoverageBed -ibam stdin  '-bg'  '-strand' '+'  '-split'  '-g' '/projects/ps-yeolab/genomes/hg19/hg19.chrom.sizes'  > /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.pos.bg  
-  samtools view -h /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.bam | awk 'BEGIN {OFS="\t"} {if(!!and($2,0x0080)) {if(!!and($2, 0x0004)) {$2 = $2 - 16} else {$2 = $2 + 16}}; print $0}'  | samtools view -bS - | genomeCoverageBed -ibam stdin  '-bg'  '-strand' '-'  '-split'  '-g' '/projects/ps-yeolab/genomes/hg19/hg19.chrom.sizes'  > /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.neg.bg  
- submit_miso_pipeline.py  '--bam' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.bam'  '--sample-id' 'SAMPLE'  '--genome' 'hg19'  '--do-not-submit'  '--output-sh' 'SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.miso.sh'  &&  'sh' 'SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.miso.sh'  &&  'touch' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.miso'  
- make_rnaseqc.py  '/DIRECTORY/SA-KO3P1_S20_L007_R1_001.polyATrim.adapterTrim.rmRep.sorted.rg.bam' '/DIRECTORY/SA-KO3P2_S13_L006_R1_001.polyATrim.adapterTrim.rmRep.sorted.rg.bam' '/DIRECTORY/SA-KOSS2_S23_L007_R1_001.polyATrim.adapterTrim.rmRep.sorted.rg.bam' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.bam' '/DIRECTORY/SA-WT3P1_S19_L007_R1_001.polyATrim.adapterTrim.rmRep.sorted.rg.bam' '/DIRECTORY/SA-WT3P2_S22_L007_R1_001.polyATrim.adapterTrim.rmRep.sorted.rg.bam' '/DIRECTORY/SA-WTSS1_S11_L006_R1_001.polyATrim.adapterTrim.rmRep.sorted.rg.bam' '/DIRECTORY/SA-WTSS2_S21_L007_R1_001.polyATrim.adapterTrim.rmRep.sorted.rg.bam'  >  '/DIRECTORY/rnaseqc_hg19.txt'  
- normalize_bedGraph.py  '--bg' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.neg.bg'  '--bam' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.bam'  >  '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.norm.neg.bg'  
- negBedGraph.py  '--bg' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.norm.neg.bg'  >  '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.norm.neg.t.bg'  
- bedGraphToBigWig /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.norm.neg.t.bg /projects/ps-yeolab/genomes/hg19/hg19.chrom.sizes /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.norm.neg.bw 
- normalize_bedGraph.py  '--bg' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.pos.bg'  '--bam' '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.bam'  >  '/DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.norm.pos.bg'  
- bedGraphToBigWig /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.norm.pos.bg /projects/ps-yeolab/genomes/hg19/hg19.chrom.sizes /DIRECTORY/SAMPLE.polyATrim.adapterTrim.rmRep.sorted.rg.norm.pos.bw 
